Microsoft.Toolkit.Uwp.UI.Controls/FlexPanel/FlexPanel.cs

```diff
@@ -0,0 +1,1572 @@
+                    }
+
+                    orderedIndices = null;
+                    if (item.ShouldOrderChildren && item.Count > 0)
+                    {
+                        var indices = new int[item.Count];
+
+                        // Creating a list of item indices sorted using the children's `order'
+                        // attribute values. We are using a simple insertion sort as we need
+                        // stability (insertion order must be preserved) and cross-platform
+                        // support. We should eventually switch to merge sort (or something
+                        // else) if the number of items becomes significant enough.
+                        for (int i = 0; i < item.Count; i++)
+                        {
+                            indices[i] = i;
+                            for (int j = i; j > 0; j--)
+                            {
+                                int prev = indices[j - 1];
+                                int curr = indices[j];
+                                if (item.children[prev].Order <= item.children[curr].Order)
+                                {
+                                    break;
+                                }
+
+                                indices[j - 1] = curr;
+                                indices[j] = prev;
+                            }
+                        }
+
+                        orderedIndices = indices;
+                    }
+
+                    LineFlexibleMainAxisSize = 0;
+                    LineFlexGrows = 0;
+                    LineFlexShrinks = 0;
+
+                    IsCrossAxisReversed = false;
+                    Wrap = item.Wrap != FlexWrap.NoWrap;
+                    if (Wrap)
+                    {
+                        if (item.Wrap == FlexWrap.Reverse)
+                        {
+                            IsCrossAxisReversed = true;
+                            LineCrossAxisPosition = CrossAxisParentSize;
+                        }
+                    }
+                    else
+                    {
+                        LineCrossAxisPosition = Vertical ? item.PaddingLeft : item.PaddingTop;
+                    }
+
+                    NeedLines = Wrap && item.AlignContent != FlexAlignContent.Start;
+                    Lines = null;
+                    LinesSizes = 0;
+                }
+
+                public FlexItem ChildAt(FlexItem item, int i) =>
+                    item.children[orderedIndices?[i] ?? i];
+
+                public void Cleanup()
+                {
+                    orderedIndices = null;
+                    Lines = null;
+                }
+            }
+
+            private static readonly List<FlexItem> Empty = new List<FlexItem>();
+
+            public List<FlexItem>.Enumerator GetEnumerator() =>
+                (children ?? Empty).GetEnumerator();
+
+            public List<FlexItem> CopyChildrenAsList() => children is null
+                ? Empty
```

Comment:
You need to return a new list here and not the reusable empty one, otherwise a consumer could potentially insert new items there and effectively break the logic for the whole class going forward.

```suggestion
                ? new List<FlexItem>()
```